components/database/lmdb/Makefile

```diff
@@ -0,0 +1,55 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2019 Rouven Weiler
+#
+
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME =				lmdb
+COMPONENT_VERSION =				0.9.23
+COMPONENT_SUMMARY =				Lightning memory-mapped database library
+COMPONENT_SRC =					lmdb-LMDB_$(COMPONENT_VERSION)
+COMPONENT_ARCHIVE =				LMDB_$(COMPONENT_VERSION).tar.gz
+COMPONENT_ARCHIVE_HASH =		sha256:abf42e91f046787ed642d9eb21812a5c473f3ba5854124484d16eadbe0aa9c81
+COMPONENT_ARCHIVE_URL =			https://github.com/LMDB/lmdb/archive/LMDB_$(COMPONENT_VERSION).tar.gz
+COMPONENT_PROJECT_URL =			https://github.com/LMDB/lmdb
+COMPONENT_FMRI =				database/lmdb
+COMPONENT_CLASSIFICATION =		System/Databases
+COMPONENT_LICENSE =				OpenLDAP Public License
+COMPONENT_LICENSE_FILE =		$(COMPONENT_NAME).license
+
+include $(WS_MAKE_RULES)/prep.mk
+include $(WS_MAKE_RULES)/justmake.mk
+include $(WS_MAKE_RULES)/ips.mk
+
+# set SOVERSION here since it is needed by all targets.
```

Comment:
1) Can we set in in COMPONENT_BUILD_ENV and COMPONENT_INSTALL_ENV, as it's a common pattern instead of exporting?
2) What are lmdb promises about ABI? If you make SOVERSION=$(COMPONENT_VERSION), even minor update will result in ABI breakage.

Reply:
> 
> 
>     1. Can we set in in COMPONENT_BUILD_ENV and COMPONENT_INSTALL_ENV, as it's a common pattern instead of exporting?
Well, I thought I got the approval by Mno-hime. You are right. Shall I now again change something?
> 
>     2. What are lmdb promises about ABI stability? If you make SOVERSION=$(COMPONENT_VERSION), even minor update will result in ABI breakage.
The thing with the versioning was done due to the info I got from Mno-hime. He sent me info how OpenSuse and Fedora do the versioning. I just compied that. Concerning the ABI stability I cannot tell you. I am a simple user...

Reply:
ABI stability looks good: https://abi-laboratory.pro/index.php?view=timeline&l=lmdb.

openSUSE uses version-based soname because they are aggressive in rebuilding, Fedora isn't that much so they stick with soname set to `0.0.0`. Maybe we should as well.

Reply:
Say I this way: tell me I change it to whatever you like. ;o) In the end I would say there is not real issue in using the complete version number and do a symlink as 0.0.0 always looks like a bug to me.

Reply:
I suggest to follow https://src.fedoraproject.org/rpms/lmdb/blob/master/f/lmdb-make.patch in this.